### PR TITLE
Fix for AbstractState bug in Marathon and Chronos

### PR DESCRIPTION
**Pull request by swartzrock**
Implemented a pure-Scala version of ZooKeeperState called ZookeeperDirectState in order to fix the crashing AbstractState bug (https://github.com/mesosphere/marathon/issues/834).

Note - after pulling, please set the version correctly and publish so that I can update Marathon and Chronos to use this new version.

### CODE REVIEW

File: src/main/proto/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Google Protocol Buffers
+
+Mesos-utils is using Google [Protocol Buffers](https://developers.google.com/protocol-buffers) to marshal and unmarshal data from the persistent store.
+
+## Dependencies
+
+- Google [Protocol Buffers](https://developers.google.com/protocol-buffers) v2.5 should be installed on the system.
+-- Note: protoc 2.6 (and higher) will generate incompatible code. Make sure your version of protoc is 2.5.
+
+## Rebuilt the protos
```
Comment (ConnorDoyle) [inline src/main/proto/README.md]: Did you mean s/Rebuilt/Rebuild ?

Comment (swartzrock) [inline src/main/proto/README.md]: Probably. Copied that from https://github.com/mesosphere/marathon/blob/master/src/main/proto/README.md

File: src/main/proto/mesos-utils.proto [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package mesosphere.mesos.protos;
+
+option java_package = "mesosphere.mesos.protos";
+option java_outer_classname = "StateProtos";
+
+// Describes a state entry, a versioned (via a UUID) key/value pair.
+// Copied from libmesos (src/messages/state.pro) to ensure that ZookeeperDirectState is
```
Comment (guenter) [inline src/main/proto/mesos-utils.proto]: Why not import the proto from Mesos?

Comment (swartzrock) [inline src/main/proto/mesos-utils.proto]: You mean from the mesos jar? It's not included in there. I could leave instructions for users to copy the file itself from ./src/messages/state.proto, is that what you mean?

**Comment by swartzrock** [comment]
Closing this and opening https://github.com/mesosphere/mesos-utils/pull/17

**Pull request closed without merging**